src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Schemas/ControlTokenDictionary.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+using System.Xml.Linq;
+using Microsoft.PowerFx.Core.Texl.Intellisense;
+using Microsoft.PowerFx.LanguageServerProtocol.Protocol;
+
+namespace Microsoft.PowerFx.LanguageServerProtocol.Schemas
+{
+    internal class ControlTokenDictionary : IEnumerable<KeyValuePair<string, ControlToken>>
```

Comment:
You do not need to implement IEnumerable of KeyValuePair. Just implement IEnumerable<ControlToken>. This way in LanguageServer.cs you won't have to do GetControlTokens(), you can just serialize the ControlTokenDictionary as IEnumerable<ControlToken>. Make things simple. Otherwise looks good to me

Reply:
I think I am going to leave this, since it helps in easy lookup.

Reply:
I will leave it up to you but it is against what's commonly done. if you get any further feedback on this then you can consider it again but feel free to leave it like this for now

Reply:
This pattern is a a bit weird...

Particularly that it's called "ControlTokenDictionary", but doesn't implement IDict* interfaces.

One approach is just:
class ControlTokens {

  public void Add(); 
  public IEnumerable<...> Tokens  {get;set; }
}

That's also gives you a path to add future other enumerations , like 
  public IEnumerable<string> TokensNames ;